crates/libp2p-networking/src/network/node.rs

```diff
@@ -177,74 +176,42 @@ impl<K: SignatureKey + 'static> NetworkNode<K> {
     #[instrument]
     pub async fn new(config: NetworkNodeConfig<K>) -> Result<Self, NetworkError> {
         // Generate a random `KeyPair` if one is not specified
-        let identity = if let Some(ref kp) = config.identity {
-            kp.clone()
-        } else {
-            Keypair::generate_ed25519()
-        };
+        let keypair = config
+            .keypair
+            .clone()
+            .unwrap_or_else(Keypair::generate_ed25519);
 
         // Get the `PeerId` from the `KeyPair`
-        let peer_id = PeerId::from(identity.public());
+        let peer_id = PeerId::from(keypair.public());
 
-        // Generate the transport from the identity, stake table, and auth message
+        // Generate the transport from the keypair, stake table, and auth message
         let transport: BoxedTransport = gen_transport::<K>(
-            identity.clone(),
+            keypair.clone(),
             config.stake_table.clone(),
             config.auth_message.clone(),
         )
         .await?;
 
         // Generate the swarm
         let mut swarm: Swarm<NetworkDef<K>> = {
-            // Use the hash of the message's contents as the ID
-            // Use blake3 for much paranoia at very high speeds
+            // Use the `Blake3` hash of the message's contents as the ID
             let message_id_fn = |message: &GossipsubMessage| {
                 let hash = blake3::hash(&message.data);
                 MessageId::from(hash.as_bytes().to_vec())
             };
 
-            let params = if let Some(ref params) = config.mesh_params {
-                params.clone()
-            } else {
-                // NOTE this should most likely be a builder pattern
-                // at some point in the future.
-                match config.node_type {
-                    NetworkNodeType::Bootstrap => MeshParams {
-                        mesh_n_high: 1000, // make this super high in case we end up scaling to 1k
-                        // nodes
-                        mesh_n_low: 10,
-                        mesh_outbound_min: 5,
-                        mesh_n: 15,
-                    },
-                    NetworkNodeType::Regular => MeshParams {
-                        mesh_n_high: 15,
-                        mesh_n_low: 8,
-                        mesh_outbound_min: 4,
-                        mesh_n: 12,
-                    },
-                    NetworkNodeType::Conductor => MeshParams {
-                        mesh_n_high: 21,
-                        mesh_n_low: 8,
-                        mesh_outbound_min: 4,
-                        mesh_n: 12,
-                    },
-                }
-            };
-
-            // Create a custom gossipsub
+            // Derive a `Gossipsub` config from our gossip config
             let gossipsub_config = GossipsubConfigBuilder::default()
```

Comment:
Right, so the (hotshot) `GossipConfig` is meant to represent values that we _probably_ want to configure. I can't pass a built (libp2p) `GossipsubConfig` around because we do some fundamental (possibly breaking) changes like the `Blake3` override for the message ID. And I can't pass around just the (libp2p) `GossipsubConfigBuilder` because it isn't cloneable, so we can't build off of it

So we take from the default (libp2p) `GossipsubConfig` and audit the defaults with the user-configurable stuff, if that makes sense